lib/source.rb

```diff
@@ -0,0 +1,17 @@
+require_relative 'item'
+
+class Source
+  attr_accessor :source
+
+  def initialize(name)
+    @id = Random.rand(1..1000)
+    @name = name
+    @items = []
+  end
+
+  #   add function
+  def add_item(item = Item.new())
```

Comment:
For example, if I want to test this code, that's how I would do so:

```

movie = Movie.new(false)
puts movie.can_be_archived?
source = Source.new('HBO')

source.add_item = (movie)
puts source.items
puts movie.source
```
So you don't need to create the item inside of the parameter and also that returns an error because you are missing the Item's required parameters, for that I think you can just change your method to something like this:

```
def add_item(item)
    @items << item
    item.source = self
end
```

Therefore you don't need to require the item, because when you pass the item you will also call the `.source` setter method defined in your Source class, that will push it into your items.